### PR TITLE
Add new input to allow users to force install dependencies, bypassing cache

### PR DESCRIPTION
**Pull request by HDash**
Tackles Issue #86 

**Problem:** When a developer adds new dependencies in the DESCRIPTION file and pushes an update without incrementing the package version, rworkflows fails to install these newly added dependencies. This happens because the cache is not updated, causing dependency issues.

**Proposed Fix:** Introduce a new input parameter, force_install. When set to true, the cache will be bypassed, and all packages will be reinstalled from scratch, ensuring new dependencies are included. The default is false, maintaining the current behaviour.

### CODE REVIEW

**Comment by bschilder** [comment]
Thanks for the upgrade @HDash , I think this is very helpful to have and I've been meaning to implement something like this for a while.

The only things I'd change is making it slightly more streamlined by passing `force_install` as an arg to the same yaml chunk, rather than duplicating the chunk. I'll make some suggestions to the code to show you what I mean.

**Comment by bschilder** [comment]
I should also mention, you can always bypass the cache by adding "[nocache]" to the commit message. But this is kind of a nuclear approach since it ignores the cache of all softwares, not just the R package deps.

File: action.yml [hunk 1]
```diff
@@ -332,6 +339,7 @@ runs:
       shell: Rscript {0} 
 
     - name: 🛠 Install dependencies pass 1 
+      if: inputs.force_install == 'false'
```
Comment (bschilder) [inline action.yml:342]: remove this and instead pass a force_install as an env variable.

File: action.yml [hunk 2]
```diff
@@ -332,6 +339,7 @@ runs:
...
       env:
         timeout: ${{ inputs.timeout }}
         GITHUB_TOKEN: ${{ inputs.GITHUB_TOKEN }}
```
Comment (bschilder) [inline action.yml:345]: add a new line below here:
```yml
force_install: ${{ inputs.force_install }}
```

File: action.yml [hunk 3]
```diff
@@ -345,6 +353,22 @@ runs:
          remotes::install_local(repos=repos, dependencies=TRUE, build_vignettes=FALSE, upgrade=TRUE, force=FALSE)
       continue-on-error: true
       shell: Rscript {0}
+
```
Comment (bschilder) [inline action.yml:356]: remove this duplicate chunk

File: action.yml [hunk 4]
```diff
@@ -345,6 +353,22 @@ runs:
          remotes::install_local(repos=repos, dependencies=TRUE, build_vignettes=FALSE, upgrade=TRUE, force=FALSE)
```
Comment (bschilder) [inline action.yml:353]: pass the env var to the `force` arg:
```
remotes::install_local(repos=repos, dependencies=TRUE, build_vignettes=FALSE, upgrade=TRUE, force=Sys.getenv("force_install"))

```

**Review by bschilder — CHANGES_REQUESTED** [review]

**Comment by HDash** [comment]
I was wondering how I could pass the input directly to `force=` within the same YAML chunk. Perhaps I can set an environment variable and use that?

**Review by bschilder — APPROVED** [review]
Looks good! thanks!

**Comment by HDash** [comment]
I had completely forgotten about `Sys.getenv`-thanks for the suggestion!
Let me know if any other changes are required.

**Pull request merged**